compiler/noirc_frontend/src/monomorphization/ast.rs

```diff
@@ -490,7 +500,7 @@ impl std::fmt::Display for Type {
                 write!(f, "fn({}) -> {}{}", args.join(", "), ret, closure_env_text)
             }
             Type::Slice(element) => write!(f, "[{element}]"),
-            Type::MutableReference(element) => write!(f, "&mut {element}"),
+            Type::Reference(element) => write!(f, "&mut {element}"),
```

Comment:
Should this be just `&`? I don't know if we have the concept of mutable reference anymore here to decide which one to print.

Reply:
It can be either I suppose. I can keep the original boolean though it'd just be used for printing. If we're considering this as a possible debug tool then it can be helpful to have the clearer output.